NPE thrown while creating Knative channel with yaks 0.3

I have a problem with step "Given create Knative channel messages" - with yaks 0.2.0-202101130033 it passes ok, but with yaks 0.3.0-202103010008 I get NPE:
Given create Knative channel messages              # org.citrusframework.yaks.knative.KnativeMessagingSteps.createChannel(java.lang.String)
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test       com.consol.citrus.exceptions.TestCaseFailedException: Test case failed
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test         at com.consol.citrus.DefaultTestCase.executeAction(DefaultTestCase.java:144)
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test         at com.consol.citrus.DefaultTestCaseRunner.run(DefaultTestCaseRunner.java:125)
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test         at com.consol.citrus.GherkinTestActionRunner.given(GherkinTestActionRunner.java:26)
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test         at org.citrusframework.yaks.knative.KnativeMessagingSteps.createChannel(KnativeMessagingSteps.java:55)
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test         at ✽.create Knative channel messages(classpath:org/citrusframework/yaks/jira-inmem-binding.feature:7)
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test Caused by: java.lang.NullPointerException
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test         at org.citrusframework.yaks.knative.actions.messaging.CreateChannelAction.doExecute(CreateChannelAction.java:51)
test-jira-inmem-binding-c0v1esro7cbrn5f8n0mg-6ktrc test         at com.consol.citrus.actions.AbstractTestAction.execute(AbstractTestAction.java:58)

--- a/java/pom.xml
+++ b/java/pom.xml
@@ -123,6 +123,12 @@
         <groupId>io.fabric8</groupId>
         <artifactId>knative-mock</artifactId>
         <version>${knative-client.version}</version>
+        <exclusions>
+          <exclusion>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+          </exclusion>
+        </exclusions>
         <scope>test</scope>
       </dependency>
 
@@ -395,9 +401,6 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-surefire-plugin</artifactId>
           <version>${maven-surefire-plugin.version}</version>
-          <configuration>
-            <failIfNoTests>false</failIfNoTests>
-          </configuration>
         </plugin>
         <plugin>
           <artifactId>maven-remote-resources-plugin</artifactId>

--- a/java/runtime/yaks-runtime-maven/pom.xml
+++ b/java/runtime/yaks-runtime-maven/pom.xml
@@ -133,8 +133,17 @@
       <plugins>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>${maven-surefire-plugin.version}</version>
+          <configuration>
+            <failIfNoTests>false</failIfNoTests>
+          </configuration>
+        </plugin>
+
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-failsafe-plugin</artifactId>
-          <version>${surefire.version}</version>
+          <version>${maven-failsafe-plugin.version}</version>
           <executions>
             <execution>
               <id>integration-tests</id>

--- a/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/KnativeEventingSteps.java
+++ b/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/KnativeEventingSteps.java
@@ -25,6 +25,7 @@ import io.cucumber.datatable.DataTable;
 import io.cucumber.java.Before;
 import io.cucumber.java.Scenario;
 import io.cucumber.java.en.Given;
+import io.fabric8.knative.client.KnativeClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import org.citrusframework.yaks.kubernetes.KubernetesSupport;
 
@@ -45,6 +46,7 @@ public class KnativeEventingSteps {
     private Citrus citrus;
 
     private KubernetesClient k8sClient;
+    private KnativeClient knativeClient;
 
     private String brokerName = KnativeSettings.getBrokerName();
 
@@ -57,6 +59,10 @@ public class KnativeEventingSteps {
         if (k8sClient == null) {
             k8sClient = KubernetesSupport.getKubernetesClient(citrus);
         }
+
+        if (knativeClient == null) {
+            knativeClient = KnativeSupport.getKnativeClient(citrus);
+        }
     }
 
     @Given("^Knative broker ([^\\s]+)$")
@@ -68,13 +74,13 @@ public class KnativeEventingSteps {
     public void createBroker(String brokerName) {
         setBrokerName(brokerName);
 
-        runner.given(knative().client(k8sClient)
+        runner.given(knative().client(k8sClient).client(knativeClient)
                 .brokers()
                 .create(brokerName));
 
         if (KnativeSteps.autoRemoveResources) {
             runner.then(doFinally()
-                    .actions(knative().client(k8sClient)
+                    .actions(knative().client(k8sClient).client(knativeClient)
                             .brokers()
                             .delete(brokerName)));
         }
@@ -85,21 +91,21 @@ public class KnativeEventingSteps {
         runner.then(repeatOnError()
             .autoSleep(500)
             .until((i, context) -> i == 10)
-            .actions(knative().client(k8sClient)
+            .actions(knative().client(k8sClient).client(knativeClient)
                     .brokers()
                     .verify(brokerName)));
     }
 
     @Given("^create Knative trigger ([^\\s]+) on service ([^\\s]+)$")
     public void createTriggerOnService(String triggerName, String serviceName) {
-        runner.given(knative().client(k8sClient)
+        runner.given(knative().client(k8sClient).client(knativeClient)
                 .trigger()
                 .create(triggerName)
                 .onService(serviceName));
 
         if (KnativeSteps.autoRemoveResources) {
             runner.then(doFinally()
-                    .actions(knative().client(k8sClient)
+                    .actions(knative().client(k8sClient).client(knativeClient)
                             .trigger()
                             .delete(triggerName)));
         }
@@ -107,7 +113,7 @@ public class KnativeEventingSteps {
 
     @Given("^create Knative trigger ([^\\s]+) on service ([^\\s]+) with filter on attributes$")
     public void createTriggerOnServiceFiltered(String triggerName, String serviceName, DataTable filterAttributes) {
-        runner.given(knative().client(k8sClient)
+        runner.given(knative().client(k8sClient).client(knativeClient)
                 .trigger()
                 .create(triggerName)
                 .onService(serviceName)
@@ -115,7 +121,7 @@ public class KnativeEventingSteps {
 
         if (KnativeSteps.autoRemoveResources) {
             runner.then(doFinally()
-                    .actions(knative().client(k8sClient)
+                    .actions(knative().client(k8sClient).client(knativeClient)
                             .trigger()
                             .delete(triggerName)));
         }
@@ -123,14 +129,14 @@ public class KnativeEventingSteps {
 
     @Given("^create Knative trigger ([^\\s]+) on channel ([^\\s]+)$")
     public void createTriggerOnChannel(String triggerName, String channelName) {
-        runner.given(knative().client(k8sClient)
+        runner.given(knative().client(k8sClient).client(knativeClient)
                 .trigger()
                 .create(triggerName)
                 .onChannel(channelName));
 
         if (KnativeSteps.autoRemoveResources) {
             runner.then(doFinally()
-                    .actions(knative().client(k8sClient)
+                    .actions(knative().client(k8sClient).client(knativeClient)
                             .trigger()
                             .delete(triggerName)));
         }
@@ -138,7 +144,7 @@ public class KnativeEventingSteps {
 
     @Given("^create Knative trigger ([^\\s]+) on channel ([^\\s]+) with filter on attributes$")
     public void createTriggerFiltered(String triggerName, String channelName, DataTable filterAttributes) {
-        runner.given(knative().client(k8sClient)
+        runner.given(knative().client(k8sClient).client(knativeClient)
                 .trigger()
                 .create(triggerName)
                 .onChannel(channelName)
@@ -146,7 +152,7 @@ public class KnativeEventingSteps {
 
         if (KnativeSteps.autoRemoveResources) {
             runner.then(doFinally()
-                    .actions(knative().client(k8sClient)
+                    .actions(knative().client(k8sClient).client(knativeClient)
                             .trigger()
                             .delete(triggerName)));
         }

--- a/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/KnativeMessagingSteps.java
+++ b/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/KnativeMessagingSteps.java
@@ -24,6 +24,7 @@ import com.consol.citrus.annotations.CitrusResource;
 import io.cucumber.java.Before;
 import io.cucumber.java.Scenario;
 import io.cucumber.java.en.Given;
+import io.fabric8.knative.client.KnativeClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import org.citrusframework.yaks.kubernetes.KubernetesSupport;
 
@@ -42,23 +43,28 @@ public class KnativeMessagingSteps {
     private Citrus citrus;
 
     private KubernetesClient k8sClient;
+    private KnativeClient knativeClient;
 
     @Before
     public void before(Scenario scenario) {
         if (k8sClient == null) {
             k8sClient = KubernetesSupport.getKubernetesClient(citrus);
         }
+
+        if (knativeClient == null) {
+            knativeClient = KnativeSupport.getKnativeClient(citrus);
+        }
     }
 
     @Given("^create Knative channel ([^\\s]+)$")
     public void createChannel(String channelName) {
-        runner.given(knative().client(k8sClient)
+        runner.given(knative().client(k8sClient).client(knativeClient)
                 .channels()
                 .create(channelName));
 
         if (KnativeSteps.autoRemoveResources) {
             runner.then(doFinally()
-                    .actions(knative().client(k8sClient)
+                    .actions(knative().client(k8sClient).client(knativeClient)
                             .channels()
                             .delete(channelName)));
         }
@@ -66,17 +72,17 @@ public class KnativeMessagingSteps {
 
     @Given("^subscribe service ([^\\s]+) to Knative channel ([^\\s]+)$")
     public void createSubscription(String serviceName, String channelName) {
-        runner.given(knative().client(k8sClient)
+        runner.given(knative().client(k8sClient).client(knativeClient)
                 .subscriptions()
-                .create(serviceName + "subscription")
+                .create(serviceName + "-subscription")
                 .onChannel(channelName)
                 .service(serviceName));
 
         if (KnativeSteps.autoRemoveResources) {
             runner.then(doFinally()
-                    .actions(knative().client(k8sClient)
+                    .actions(knative().client(k8sClient).client(knativeClient)
                             .subscriptions()
-                            .delete(serviceName + "subscription")));
+                            .delete(serviceName + "-subscription")));
         }
     }
 }

--- a/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/ReceiveEventSteps.java
+++ b/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/ReceiveEventSteps.java
@@ -56,6 +56,7 @@ public class ReceiveEventSteps {
     public void before(Scenario scenario) {
         kubernetesSteps = new KubernetesSteps();
         CitrusAnnotations.injectAll(kubernetesSteps, citrus, context);
+        CitrusAnnotations.injectTestRunner(kubernetesSteps, runner);
         kubernetesSteps.before(scenario);
         kubernetesSteps.configureTimeout(KnativeSettings.getEventConsumerTimeout());
         kubernetesSteps.setServiceName(KnativeSettings.getServiceName());

--- a/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/actions/KnativeActionBuilder.java
+++ b/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/actions/KnativeActionBuilder.java
@@ -18,6 +18,7 @@
 package org.citrusframework.yaks.knative.actions;
 
 import com.consol.citrus.TestActionBuilder;
+import io.fabric8.knative.client.KnativeClient;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import org.citrusframework.yaks.knative.actions.eventing.CreateBrokerAction;
 import org.citrusframework.yaks.knative.actions.eventing.CreateTriggerAction;
@@ -33,6 +34,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
 
     /** Kubernetes client */
     private KubernetesClient kubernetesClient;
+    private KnativeClient knativeClient;
 
     private AbstractKnativeAction.Builder<? extends KnativeAction, ?> delegate;
 
@@ -50,6 +52,15 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
      */
     public KnativeActionBuilder client(KubernetesClient kubernetesClient) {
         this.kubernetesClient = kubernetesClient;
+        return this;
+    }
+
+    /**
+     * Use a custom Knative client.
+     * @param knativeClient
+     */
+    public KnativeActionBuilder client(KnativeClient knativeClient) {
+        this.knativeClient = knativeClient;
         return this;
     }
 
@@ -91,6 +102,10 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         if (kubernetesClient != null) {
             delegate.client(kubernetesClient);
         }
+
+        if (knativeClient != null) {
+            delegate.client(knativeClient);
+        }
         return delegate.build();
     }
 
@@ -107,6 +122,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public CreateChannelAction.Builder create(String channelName) {
             CreateChannelAction.Builder builder = new CreateChannelAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .name(channelName);
             delegate = builder;
             return builder;
@@ -119,6 +135,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public DeleteKnativeResourceAction.Builder delete(String channelName) {
             DeleteKnativeResourceAction.Builder builder = new DeleteKnativeResourceAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .component("messaging")
                     .kind("channels")
                     .name(channelName);
@@ -135,6 +152,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public CreateSubscriptionAction.Builder create(String subscriptionName) {
             CreateSubscriptionAction.Builder builder = new CreateSubscriptionAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .name(subscriptionName);
             delegate = builder;
             return builder;
@@ -147,6 +165,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public DeleteKnativeResourceAction.Builder delete(String subscriptionName) {
             DeleteKnativeResourceAction.Builder builder = new DeleteKnativeResourceAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .component("messaging")
                     .kind("subscriptions")
                     .name(subscriptionName);
@@ -163,6 +182,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public CreateTriggerAction.Builder create(String triggerName) {
             CreateTriggerAction.Builder builder = new CreateTriggerAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .name(triggerName);
             delegate = builder;
             return builder;
@@ -175,6 +195,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public DeleteKnativeResourceAction.Builder delete(String triggerName) {
             DeleteKnativeResourceAction.Builder builder = new DeleteKnativeResourceAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .component("eventing")
                     .kind("triggers")
                     .name(triggerName);
@@ -191,6 +212,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public CreateBrokerAction.Builder create(String brokerName) {
             CreateBrokerAction.Builder builder = new CreateBrokerAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .name(brokerName);
             delegate = builder;
             return builder;
@@ -203,6 +225,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public DeleteKnativeResourceAction.Builder delete(String brokerName) {
             DeleteKnativeResourceAction.Builder builder = new DeleteKnativeResourceAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .component("eventing")
                     .kind("brokers")
                     .name(brokerName);
@@ -217,6 +240,7 @@ public class KnativeActionBuilder implements TestActionBuilder.DelegatingTestAct
         public VerifyBrokerAction.Builder verify(String brokerName) {
             VerifyBrokerAction.Builder builder = new VerifyBrokerAction.Builder()
                     .client(kubernetesClient)
+                    .client(knativeClient)
                     .name(brokerName);
             delegate = builder;
             return builder;

--- a/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/actions/eventing/VerifyBrokerAction.java
+++ b/java/steps/yaks-knative/src/main/java/org/citrusframework/yaks/knative/actions/eventing/VerifyBrokerAction.java
@@ -47,7 +47,8 @@ public class VerifyBrokerAction extends AbstractKnativeAction {
             if (broker.getStatus() != null &&
                     broker.getStatus().getConditions() != null &&
                     broker.getStatus().getConditions().stream()
-                            .anyMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))) {
+                            .anyMatch(condition -> condition.getType().equals("Ready") &&
+                                    condition.getStatus().equalsIgnoreCase("True"))) {
                 LOG.info(String.format("Knative broker %s is ready", brokerName));
             } else {
                 throw new ValidationException(String.format("Knative broker '%s' is not ready", brokerName));

--- a/java/steps/yaks-knative/src/test/java/org/citrusframework/yaks/knative/KnativeServiceConfiguration.java
+++ b/java/steps/yaks-knative/src/test/java/org/citrusframework/yaks/knative/KnativeServiceConfiguration.java
@@ -84,8 +84,7 @@ public class KnativeServiceConfiguration {
                 Assertions.assertThat(message.getHeader("Ce-Subject")).isEqualTo("hello");
                 Assertions.assertThat(message.getHeader("Ce-Type")).isEqualTo("greeting");
                 Assertions.assertThat(message.getHeader("Ce-Source")).isEqualTo("https://github.com/citrusframework/yaks");
-                Assertions.assertThat(message.getHeader("Ce-Time")).isNotNull();
-                Assertions.assertThat(message.getHeader("Content-Type").toString()).isEqualTo(MediaType.APPLICATION_JSON_VALUE);
+                Assertions.assertThat(message.getHeader("Content-Type").toString()).isEqualTo(MediaType.APPLICATION_JSON_UTF8_VALUE);
                 Assertions.assertThat(message.getPayload(String.class)).isEqualTo("{\"msg\": \"Hello Knative!\"}");
 
                 return new HttpMessage().status(HttpStatus.ACCEPTED);

--- a/java/steps/yaks-knative/src/test/java/org/citrusframework/yaks/knative/KnativeTestSteps.java
+++ b/java/steps/yaks-knative/src/test/java/org/citrusframework/yaks/knative/KnativeTestSteps.java
@@ -28,7 +28,10 @@ import io.cucumber.datatable.DataTable;
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.fabric8.knative.client.KnativeClient;
-import io.fabric8.knative.eventing.v1.Trigger;
+import io.fabric8.knative.eventing.v1.Broker;
+import io.fabric8.knative.eventing.v1.BrokerStatus;
+import io.fabric8.knative.eventing.v1.TriggerList;
+import io.fabric8.knative.internal.pkg.apis.Condition;
 import io.fabric8.knative.messaging.v1.SubscriptionList;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import org.assertj.core.api.Assertions;
@@ -61,6 +64,34 @@ public class KnativeTestSteps {
                         .message(eventRequest));
     }
 
+    @Given("^activate Knative broker ([^\\s]+)$")
+    public void activateBroker(String brokerName) {
+        runner.run(new KnativeTestAction() {
+            @Override
+            public void doExecute(TestContext context) {
+                Broker broker = getKnativeClient()
+                        .brokers()
+                        .inNamespace(namespace(context))
+                        .withName(brokerName)
+                        .get();
+
+                Assertions.assertThat(broker).isNotNull();
+
+                BrokerStatus status = new BrokerStatus();
+                Condition ready = new Condition();
+                ready.setType("Ready");
+                ready.setStatus("True");
+                status.getConditions().add(ready);
+                broker.setStatus(status);
+
+                getKnativeClient()
+                        .brokers()
+                        .inNamespace(namespace(context))
+                        .updateStatus(broker);
+            }
+        });
+    }
+
     @Then("^verify test event accepted$")
     public void verifyEventAccepted() {
         runner.run(http().client("http://localhost:${knativeServicePort}/")
@@ -73,11 +104,15 @@ public class KnativeTestSteps {
         runner.run(new KnativeTestAction() {
             @Override
             public void doExecute(TestContext context) {
-                Assertions.assertThat(getKnativeClient()
+                TriggerList triggers = getKnativeClient()
                         .triggers()
                         .inNamespace(namespace(context))
-                        .withName(triggerName)
-                        .get()).isNotNull();
+                        .list();
+
+                Assertions.assertThat(triggers).isNotNull();
+                Assertions.assertThat(triggers.getItems()).isNotEmpty();
+                Assertions.assertThat(triggers.getItems().size()).isEqualTo(1);
+                Assertions.assertThat(triggers.getItems().get(0).getMetadata().getName()).isEqualTo(triggerName);
             }
         });
     }
@@ -87,17 +122,19 @@ public class KnativeTestSteps {
         runner.run(new KnativeTestAction() {
             @Override
             public void doExecute(TestContext context) {
-                Trigger trigger = getKnativeClient()
+                TriggerList triggers = getKnativeClient()
                         .triggers()
                         .inNamespace(namespace(context))
-                        .withName(triggerName)
-                        .get();
+                        .list();
 
-                Assertions.assertThat(trigger).isNotNull();
-                Assertions.assertThat(trigger.getSpec().getFilter().getAttributes().size()).isEqualTo(filter.height());
+                Assertions.assertThat(triggers).isNotNull();
+                Assertions.assertThat(triggers.getItems()).isNotEmpty();
+                Assertions.assertThat(triggers.getItems().size()).isEqualTo(1);
+                Assertions.assertThat(triggers.getItems().get(0).getMetadata().getName()).isEqualTo(triggerName);
+                Assertions.assertThat(triggers.getItems().get(0).getSpec().getFilter().getAttributes().size()).isEqualTo(filter.height());
                 filter.asMap(String.class, String.class).forEach((key, value) -> {
-                    Assertions.assertThat(trigger.getSpec().getFilter().getAttributes()).containsKey(key.toString());
-                    Assertions.assertThat(trigger.getSpec().getFilter().getAttributes().get(key.toString())).isEqualTo(value);
+                    Assertions.assertThat(triggers.getItems().get(0).getSpec().getFilter().getAttributes()).containsKey(key.toString());
+                    Assertions.assertThat(triggers.getItems().get(0).getSpec().getFilter().getAttributes().get(key.toString())).isEqualTo(value);
                 });
             }
         });

--- a/java/steps/yaks-knative/src/test/resources/org/citrusframework/yaks/knative/knative.event.consumer.feature
+++ b/java/steps/yaks-knative/src/test/resources/org/citrusframework/yaks/knative/knative.event.consumer.feature
@@ -5,6 +5,7 @@ Feature: Knative event consumer
     And Knative service port 8181
     And Knative service "hello-service"
     And Knative event consumer timeout is 5000 ms
+    And create Knative event consumer service hello-service
     And variable id is "citrus:randomNumber(4)"
     And create test event
     """
@@ -15,6 +16,7 @@ Feature: Knative event consumer
       "subject" : "hello",
       "id" : "say-hello-${id}",
       "datacontenttype" : "application/json",
+      "time": "citrus:currentDate('yyyy-MM-dd'T'HH:mm:ss')",
       "data" : "{\"msg\": \"Hello Knative!\"}"
     }
     """
@@ -26,6 +28,7 @@ Feature: Knative event consumer
       | source          | https://github.com/citrusframework/yaks |
       | subject         | hello |
       | id              | say-hello-${id} |
+      | time            | @matchesDatePattern('yyyy-MM-dd'T'HH:mm:ss')@ |
       | datacontenttype | application/json;charset=UTF-8 |
       | data            | {"msg": "Hello Knative!"} |
     Then verify test event accepted
@@ -47,6 +50,7 @@ Feature: Knative event consumer
       | Ce-Source          | https://github.com/citrusframework/yaks |
       | Ce-Subject         | hello |
       | Ce-Id              | say-hello-${id} |
+      | Ce-Time            | @matchesDatePattern('yyyy-MM-dd'T'HH:mm:ss')@ |
       | Content-Type       | application/json;charset=UTF-8 |
     Then verify test event accepted
 

--- a/java/steps/yaks-knative/src/test/resources/org/citrusframework/yaks/knative/knative.event.producer.feature
+++ b/java/steps/yaks-knative/src/test/resources/org/citrusframework/yaks/knative/knative.event.producer.feature
@@ -12,6 +12,7 @@ Feature: Knative event producer
       | source          | https://github.com/citrusframework/yaks |
       | subject         | hello |
       | id              | say-hello |
+      | time            | citrus:currentDate('yyyy-MM-dd'T'HH:mm:ss') |
       | datacontenttype | application/json |
       | data            | {"msg": "Hello Knative!"} |
 
@@ -31,6 +32,7 @@ Feature: Knative event producer
       | Ce-Source          | https://github.com/citrusframework/yaks |
       | Ce-Subject         | hello |
       | Ce-Id              | say-hello |
+      | Ce-Time            | citrus:currentDate('yyyy-MM-dd'T'HH:mm:ss') |
       | Content-Type       | application/json |
 
   Scenario: Send multiline event data
@@ -53,6 +55,7 @@ Feature: Knative event producer
       "source" : "https://github.com/citrusframework/yaks",
       "subject" : "hello",
       "id" : "say-hello",
+      "time" : "citrus:currentDate('yyyy-MM-dd'T'HH:mm:ss')",
       "datacontenttype" : "application/json",
       "data" : "{\"msg\": \"Hello Knative!\"}"
     }

--- a/java/steps/yaks-knative/src/test/resources/org/citrusframework/yaks/knative/knative.eventing.feature
+++ b/java/steps/yaks-knative/src/test/resources/org/citrusframework/yaks/knative/knative.eventing.feature
@@ -1,11 +1,13 @@
 Feature: Knative eventing
 
   Background:
+    Given Disable auto removal of Knative resources
     Given Knative namespace event-example
     And Knative broker my-broker
 
   Scenario: Create broker
     Given create Knative broker my-broker
+    When activate Knative broker my-broker
     Then Knative broker my-broker is running
 
   Scenario: Create service and trigger

--- a/java/steps/yaks-knative/src/test/resources/org/citrusframework/yaks/knative/knative.messaging.feature
+++ b/java/steps/yaks-knative/src/test/resources/org/citrusframework/yaks/knative/knative.messaging.feature
@@ -1,6 +1,7 @@
 Feature: Knative messaging
 
   Background:
+    Given Disable auto removal of Knative resources
     Given Knative namespace event-example
 
   Scenario: Create channel

